iGCS/Models/Craft/GCSCraftNotifications.m

```diff
@@ -24,4 +25,18 @@ + (void)didNavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
     });
 }
 
++ (void)didMavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
+                          andNewHeartbeat:(mavlink_heartbeat_t) newHeartbeat {
+
+    if (lastHeartbeat.base_mode != 0 &&
+        (lastHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED) == (newHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED)) {
```

Comment:
Happy to push forward with a heartbeat object but I'm not sure this is the branch to do it in and I can't really DRY it up in the mixin easily but I do see it's being used in one other spot for sure.